tests/ssr.test.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect } from 'react'
+import { act, screen } from '@testing-library/react'
+import { renderToString } from 'react-dom/server'
+import { describe, expect, it, vi } from 'vitest'
+import { create } from 'zustand'
+
+interface BearStoreState {
+  bears: number
+}
+
+interface BearStoreAction {
+  increasePopulation: () => void
+}
+
+const initialState = { bears: 0 }
+const useBearStore = create<BearStoreState & BearStoreAction>((set) => ({
+  ...initialState,
+  increasePopulation: () => set(({ bears }) => ({ bears: bears + 1 })),
+}))
+
+function Counter() {
+  const { bears, increasePopulation } = useBearStore(
+    ({ bears, increasePopulation }) => ({
+      bears,
+      increasePopulation,
+    })
+  )
+
+  useEffect(() => {
+    increasePopulation()
+  }, [increasePopulation])
+
+  return <div>bears: {bears}</div>
+}
+
+describe.skipIf(!React.version.startsWith('18'))(
+  'ssr behavior with react 18',
+  async () => {
+    it('should handle different states between server and client correctly', async () => {
+      const {
+        hydrateRoot,
+      }: {
+        hydrateRoot: (
+          container: Element | Document,
+          initialChildren: React.ReactNode
+        ) => void
+      } = await vi.importActual('react-dom/client')
```

Comment:
```suggestion
      const { hydrateRoot } = await vi.importActual<typeof import('react-dom/client')>('react-dom/client')
```

Reply:
Thank you for the suggestion. I've made the requested changes as you suggested.